crates/ra_project_model/src/cargo_workspace.rs

```diff
@@ -141,12 +142,14 @@ impl CargoWorkspace {
 
         for meta_pkg in meta.packages {
             let is_member = ws_members.contains(&meta_pkg.id);
+            let name = meta_pkg.name;
```

Comment:
I think you can destructure meta_pkg to avoid cloning: `let MetaPkg { name, edition } = meta_pkg`.

Reply:
But I have to clone it if I'm also using `name` in the `unwrap_or_else` closure's panic message. Perhaps mentioning the package's name in the panic msg is not necessary? Or silently reverting to a default edition rather than panicking would be better?

Reply:
Sorry, I was typing on my phone. You can make it work like this:

```rs
            let cargo_metadata::Package { name, manifest_path, edition, .. } = meta_pkg;
            let edition = Edition::from_str(&edition)
                .unwrap_or_else(|e| panic!("unknown edition {} for package {:?}", e, name));
            let pkg = packages.alloc(PackageData {
                name,
                manifest: manifest_path,
                targets: Vec::new(),
                is_member,
                edition,
                dependencies: Vec::new(),
                features: Vec::new(),
            });
```

I don't have any strong opinion wrt. handling the error, but the panicking is probably not ideal.

Reply:
Nice! 
Regarding panicking: 
- `cargo_workspace::from_cargo_metadata()` returns a `Result<CargoWorkspace>` which means I can `map_err( ... )?;` instead to bubble the error up.
- `fixture::parse_meta()` returns `ParsedMeta`, not panicking might require a default value or changing to Result<ParsedMeta>